src/server/generic_family.cc

```diff
@@ -28,6 +31,59 @@ using namespace std;
 using namespace facade;
 
 namespace {
+using VersionBuffer = std::array<char, 2>;
+using CrcBuffer = std::array<char, sizeof(uint64_t)>;
+
+VersionBuffer MakeRdbVersion() {
+  VersionBuffer buf;
+  buf[0] = RDB_VERSION & 0xff;
+  buf[1] = (RDB_VERSION >> 8) & 0xff;
+  return buf;
+}
+
+CrcBuffer MakeCheckSum(std::string_view dump_res) {
+  uint64_t chksum = crc64(0, reinterpret_cast<const uint8_t*>(dump_res.data()), dump_res.size());
+  CrcBuffer buf;
+  absl::little_endian::Store64(buf.data(), chksum);
+  return buf;
+}
+
+std::string AppendFooter(std::string_view dump_res) {
```

Comment:
It's called `AppendFooter` but it does not really append.
I prefer having a consistent signature like `void AppendFooter(string* msg);`